cpp/llm_chat.cc

```diff
@@ -989,14 +616,15 @@ class LLMChat {
     }
   }
 
-  void UpdateLogitsOrProbOnCPU(NDArray logits_or_prob) {
+  void UpdateLogitsOrProbOnCPUSync(NDArray logits_or_prob) {
     if (!logits_on_cpu_.defined()) {
       logits_on_cpu_ = logits_or_prob.CopyTo(DLDevice{kDLCPU, 0});
     } else {
       ICHECK_EQ(logits_on_cpu_->shape[0], logits_or_prob->shape[0])
           << "Expect size of logits remain unchanged";
       logits_on_cpu_.CopyFrom(logits_or_prob);
     }
+    TVMSynchronize(device_.device_type, device_.device_id, nullptr);
```

Comment:
We only need this when device is not CPU, is that correct?

Reply:
That is true, when device is cpu it is no op